src/vs/editor/test/common/core/lineTokens.test.ts

```diff
@@ -42,6 +42,31 @@ suite('LineTokens', () => {
 		);
 	}
 
+	function renderLineTokens(tokens: LineTokens): string {
+		let result = '';
+		const str = tokens.getLineContent();
+		let lastOffset = 0;
+		for (let i = 0; i < tokens.getCount(); i++) {
+			result += str.substring(lastOffset, tokens.getEndOffset(i));
+			result += `(${tokens.getMetadata(i)})`;
+			lastOffset = tokens.getEndOffset(i);
+		}
+		return result;
+	}
+
+	test('withInserted', () => {
```

Comment:
:+1: Would also be interesting to have a test for some edge cases, like inserting at the last offset or inserting multiple injected texts at the same offset and that the order is maintained.